ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -534,15 +537,17 @@ private class JdbcLedgerDao(
                   for (submitter <- tx.submittingParty; appId <- tx.applicationId;
                     cmdId <- tx.commandId)
                     yield SubmitterInfo(submitter, appId, cmdId, Instant.EPOCH)
-                transactionsWriter.write(
-                  submitterInfo = submitterInfo,
-                  workflowId = tx.workflowId,
-                  transactionId = tx.transactionId,
-                  ledgerEffectiveTime = tx.ledgerEffectiveTime,
-                  offset = offset,
-                  transaction = tx.transaction.mapNodeId(splitOrThrow),
-                  divulgedContracts = Nil,
-                )
+                transactionsWriter
+                  .prepare(
+                    submitterInfo = submitterInfo,
+                    workflowId = tx.workflowId,
+                    transactionId = tx.transactionId,
+                    ledgerEffectiveTime = tx.ledgerEffectiveTime,
+                    offset = offset,
+                    transaction = tx.transaction.mapNodeId(splitOrThrow),
+                    divulgedContracts = Nil,
+                  )
```

Comment:
Can't you pull this preparation up too?

Reply:
We can evaluate this but for the time being I'd leave this year, it only affects Sandbox Classic and the scenario loader, which is not really performance sensitive. But in general I agree it would make sense.

Reply:
Sorry, I haven't given enough context: the problem is that we would need to add a non-trivial bit of code to handle this case specifically and I would prefer to keep this sub-optimal but without any additional infrastructure, unless we really see there is a use for it.